File: leetcode-solution/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic={}
        for i,n in enumerate(nums):
            diff=target-n
            if n in dic:
                return [dic[n],i]
            else:
                dic[diff]=i

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer , Integer> Dict = new HashMap();

        for(int i =0 ; i< nums.length; i++){

            if (Dict.containsKey( (target - nums[i]))){
                return new int[] {Dict.get( (target - nums[i])), i };
            } 
            Dict.put(nums[i], i);
        }

        throw new IllegalArgumentException("No two sum solution");
    
    }
}